### PR TITLE
[2.19.x] G-9247 Loosing Result Selector Menu

### PR DESCRIPTION
**Pull request by abel-connexta**
ddf-ui PRs:
3.4.x codice/ddf-ui#428

---------------------
#### What does this PR do?
Resolved following List issues:
Issue 1 - Error when saving "NOT AND" or "NOT OR" filters.
Issue 2 - Error when remove all fields in filter and saving.

#### Who is reviewing it? 

@andrewzimmer 
@hayleynorton
@cassandrabailey293 
@zta6 
@Lambeaux 

#### How should this be tested?

**Issue 1 - Error when saving "NOT AND" or "NOT OR" filters.**
1. Go to Workspaces, open or click create a New Workspace
2. Perform any type of Search that returns results
3. Click on "Filter" Result menu icon 
4.  Select "NOT AND" or "NOT OR"
5.  Open console log
6. Add some fields and click "Save Filter"
7. Ensure no error occurs
8. Ensure the Result menu icons are still there and able to edit filter created
9. Refresh page and ensure no error occurs and Result menu icons are still there and able to edit filter created  
10. Ensure no regression with Filters

**Issue 2 - Error when saving remove all fields in filter and saving.**
Follow steps 1 to 3 from Issue 1.

4.  Click minus icon to remove  filter field(s)
5.  Open console log
6. Click "Save Filter"
7. Ensure no error occurs, and Ensure when re-open "Filter" it is reset to default.
8. Add some field(s) to the filter, then Save Filter.
9. Refresh page and follow steps 1 to 7.

#### What are the relevant tickets?
Fixes: #____

##### Issue 1 - Before
Before-GSR-9247_i1

##### Issue 1 - After
After-GSR-9247_i1

##### Issue 2 - Before
Before-GSR-9247_i2

##### Issue 2 - After
After-GSR-9247_i2

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/js/model/User.js [hunk 1]
```diff
@@ -205,6 +206,13 @@ User.Preferences = Backbone.AssociatedModel.extend({
     this.listenTo(this, 'change:goldenLayoutAlert', this.savePreferences)
     this.listenTo(this, 'change:mapHome', this.savePreferences)
   },
+  get(attr) {
```
Comment (cassandrabailey293) [inline ui/packages/catalog-ui-search/src/main/webapp/js/model/User.js:209]: ❓ so does this get function override the default get method? in other words, does this method get called every time we do `user.get(...)` ?

Comment (abel-connexta) [reply ui/packages/catalog-ui-search/src/main/webapp/js/model/User.js:209]: this get function override the default get method for `preferences`, so when we do `user.get('user').get('preferences').get('resultFilter')`, so in this case would be the `.get('resultFilter')`

Comment (cassandrabailey293) [reply ui/packages/catalog-ui-search/src/main/webapp/js/model/User.js:209]: ok, as long as it preserves the behavior of `user.get('user').get('preferences').get(...)` for the non- result filter cases, then we're good

Comment (abel-connexta) [reply ui/packages/catalog-ui-search/src/main/webapp/js/model/User.js:209]: Yep it does, line 210 gets attributes as it usually does, and only alters data for 'resultFilter'. https://github.com/codice/ddf/blob/bc9c874dec1764ad8b41f89e80c36278fc7012ba/ui/packages/catalog-ui-search/src/main/webapp/js/model/User.js#L210

**Review by cassandrabailey293 — APPROVED** [review]

**Review by mojogitoverhere — APPROVED** [review]

**Review by shaundmorris — APPROVED** [review]

**Comment by shaundmorris** [comment]
build now

**Comment by cassandrabailey293** [comment]
hero started

**Comment by cassandrabailey293** [comment]
hero successful 🎉 
verified NOT AND and NOT OR are working without throwing errors in the result filter. and that getting rid of the filter terms and saving resets the filter form to its default. Refreshed after setting, resetting, removing, and saving without a problem. LGTM!

**Pull request merged**